vtrace/platforms/win32.py

```diff
@@ -2210,3 +2223,116 @@ def getCacheSyms(self):
             ret.append( (addr-self.loadbase, size, name, stype) )
         return ret
 
+    def genSymbols(self, symopts=None, symtypes=None):
+        '''
+        Generator of symbols... does not cache symbols.  This is good for PDBs of unusual size (POUSes).
+        '''
```

Comment:
any time you hand in a callback, it can be a function or a bound method. The caller needn't provide "self" since it's bound. ( so it should work great ).  That being said, the callback implementation can easily be used to create other versions of the API ( similar to how you can build a sync API on top of an async API, but not an async API on top of a sync API ).  For example, 
```
x = []
thing.doStuffWithCallback( x.append )
return x
```
So, starting with the callback impl at the lowest level, you get your optimized case, and the easy ability to provide "simpler" api variants.  Even a yield generator if you're willing to fire a thread which feeds a queue ( that the original calling thread yields from ).